src/components/CoinList/CoinList.js

```diff
@@ -1,83 +1,85 @@
-            const pcp24h = round(price_change_percentage_24h_in_currency, 1);
-            const pcp7d = round(price_change_percentage_7d_in_currency, 1);
-            const totalVolume = round(total_volume, 0);
-            const marketCap = round(market_cap, 0);
-            return (
-              <CoinRow
-                key={symbol}
-                id={id}
-                symbol={symbol}
-                name={name}
-                market_cap={marketCap}
-                current_price={currentPrice}
-                price_change_percentage_24h_in_currency={pcp24h}
-                price_change_percentage_1h_in_currency={pcp1h}
-                price_change_percentage_7d_in_currency={pcp7d}
-                total_volume={totalVolume}
-                image={image}
-                market_cap_rank={market_cap_rank}
-                url={url}
-              />
-            );
-          })}
-        </UL>
+        {isLoading && <div style={{ color: 'white' }}>Loading...</div>}
+        {dataReady && (
+          <UL>
+            <LI>
+              <P>Rank</P>
+              <P>Coin</P>
+              <P>Price</P>
+              <P>1hr</P>
+              <P>24hr</P>
+              <P>7 days</P>
+              <P>24hr Volume</P>
+              <P>Market Cap</P>
+              <P>Last 7 Days</P>
+            </LI>
+            {coinList.map((coin) => {
+              const {
+                id,
+                symbol,
+                name,
+                market_cap,
+                current_price,
+                image,
+                price_change_percentage_1h_in_currency,
+                price_change_percentage_24h_in_currency,
+                price_change_percentage_7d_in_currency,
+                total_volume,
+                market_cap_rank,
+                url,
+              } = coin;
+              const currentPrice = round(current_price, 2);
+              const pcp1h = round(price_change_percentage_1h_in_currency, 1);
+              const pcp24h = round(price_change_percentage_24h_in_currency, 1);
+              const pcp7d = round(price_change_percentage_7d_in_currency, 1);
+              const totalVolume = round(total_volume, 0);
+              const marketCap = round(market_cap, 0);
+              return (
+                <CoinRow
+                  key={symbol}
+                  id={id}
+                  symbol={symbol}
+                  name={name}
+                  market_cap={marketCap}
+                  current_price={currentPrice}
+                  price_change_percentage_24h_in_currency={pcp24h}
+                  price_change_percentage_1h_in_currency={pcp1h}
+                  price_change_percentage_7d_in_currency={pcp7d}
+                  total_volume={totalVolume}
+                  image={image}
+                  market_cap_rank={market_cap_rank}
+                  url={url}
+                />
+              );
+            })}
```

Comment:
```
{coinList.map((coin) => {
              const {
                market_cap,
                current_price,
                price_change_percentage_1h_in_currency,
                price_change_percentage_24h_in_currency,
                price_change_percentage_7d_in_currency,
                total_volume,
                url,
              } = coin;
              const currentPrice = round(current_price, 2);
              const pcp1h = round(price_change_percentage_1h_in_currency, 1);
              const pcp24h = round(price_change_percentage_24h_in_currency, 1);
              const pcp7d = round(price_change_percentage_7d_in_currency, 1);
              const totalVolume = round(total_volume, 0);
              const marketCap = round(market_cap, 0);
              return (
                <CoinRow
                  key={coin.symbol}
                  {...coin}
                  marketCap={marketCap}
                  current_price={currentPrice}
                  pcp1h={pcp24h}
                  pcp1h={pcp1h}
                  pcp7d={pcp7d}
                  totalVolume={totalVolume}
                />
              );
            })}
`

Reply:
Couldn't I just do 
<CoinRow key={coin.symbol} {...coin} />
And then do the destructuring and rounding in the CoinRow component?
It would make the CoinList much cleaner, I think.